instrumentation/netty/netty-3.8/src/main/java/io/opentelemetry/instrumentation/auto/netty/v3_8/NettyChannelInstrumentation.java

```diff
@@ -76,16 +77,17 @@ public NettyChannelInstrumentation() {
   public static class ChannelConnectAdvice extends AbstractNettyAdvice {
     @Advice.OnMethodEnter
     public static void addConnectContinuation(@Advice.This Channel channel) {
-      Span span = NettyHttpServerTracer.TRACER.getCurrentSpan();
+      Context context = Context.current();
+      Span span = getSpan(context);
       if (span.getContext().isValid()) {
         ContextStore<Channel, ChannelTraceContext> contextStore =
             InstrumentationContext.get(Channel.class, ChannelTraceContext.class);
 
         if (contextStore
                 .putIfAbsent(channel, ChannelTraceContext.Factory.INSTANCE)
-                .getConnectionContinuation()
+                .getConnectionContext()
             == null) {
-          contextStore.get(channel).setConnectionContinuation(span);
+          contextStore.get(channel).setConnectionContext(context);
```

Comment:
Reviewing this made me realize how nervous our propagation patterns make me. Sometimes we propagate the invocation context to remount for an async callback, sometimes the context of the span itself. Makes my head spin 🤯  Don't have a good answer but just food for thought.